lib/perl/Genome/Model/Tools/DetectVariants2/SpeedseqSv.pm

```diff
@@ -0,0 +1,132 @@
+#SpeedseqSv.pm
+
+
+package Genome::Model::Tools::DetectVariants2::SpeedseqSv;
+use warnings;
+use strict;
+
+use Genome;
+use File::Basename;
+use IPC::System::Simple;
+use Genome::File::Tsv;
+
+class Genome::Model::Tools::DetectVariants2::SpeedseqSv {
+    is => 'Genome::Model::Tools::DetectVariants2::Detector',
+    has_optional => [
+        _legend_output => {
+            calculate_from => ['_temp_staging_directory'],
+            calculate => q{ File::Spec->join($_temp_staging_directory, 'legend.tsv'); },
+        },
+    ]
+};
+
+	my $temp_directory = Genome::Sys->create_temp_directory();
+	my $pkg = 'Genome::Model::Tools::Speedseq::Sv';
+
+# For the Parameters what I need to do is make two hashes.
+# The first one will have the name of the command in the SV class and the letter calling it.  
+# The second Hash will have the class string name and the value wil be either a boolean or a file location.
+	my %ref_library = (
+		B => 'full_bam_file',
+		S => 'split_read_bam_file',
+		D => 'discordant_read_bam_file',
+		R => 'reference_fasta',
+		g => 'genotype_svtyper',
+		d => 'CNVnator_read_depth',
+		A => 'annotate_vcf',
+		k => 'keep_temp_files',
+		#End of Boolean files#
+		t => 'threads',
+		x => 'exclude_bed_file',
+		m => 'min_sample_weight',
+		r => 'trim_threshold',
+		T => 'temp_directory',	
```

Comment:
I don't think we should expose the temp_directory parameter at this level - we should just make one using Genome::Sys->create_temp_directory inside of the _detect_variants subroutine and pass that in